src/lib/plugins/live-row/live-row.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, OnInit, Input, ChangeDetectionStrategy, NgZone } from '@angular/core';
+import { PluginService } from '../plugin.service';
+import { Fastdom } from '../../../core/services/fastdom';
+import { GRID_PREFIX } from 'ng2-qgrid/core/definition';
+
+@Component({
+	selector: 'q-grid-live-rows',
+	template: '<ng-content></ng-content>',
+	providers: [PluginService],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LiveRowComponent implements OnInit {
+	private currentRows: any[];
+	private animations = [];
+
+	@Input('duration') duration: number;
+
+	constructor(private plugin: PluginService, private zone: NgZone) { }
+
+	ngOnInit() {
+		const { model } = this.plugin;
+		model.animation({
+			apply: (memo, context, complete) => {
+
+				if (!this.currentRows || !memo.rows) {
```

Comment:
remove new line

it will be more readable, in next lines
```typescript
const previousRows = this.currentRows;
const currentRows = memo.rows;
this.currentRows = currentRows;
```